### ISSUE TITLE
Update Connection Error

### ISSUE THREAD

**Issue by groveld** [opnsense/core#308]
After the last update, OPNsense 15.7.7-amd64, i get an error when checking for new updates..
if i click the "Check for updates" link in the dashbvoard view, i get an "Connection Error" message.
i don't know if this is a known issue on your end, but you probably wanted to know about it.

i am using a apu1d4 with OPNsense 15.7.7-amd64.

**Comment by fichtner** [comment]
/usr/local/etc/pkg/repos/origin.conf wasn't properly written. We've seen this happen on some occasions after the update, but are not sure why. Reboot fixes this, or go to "System: Settings: General" and simply trigger "Save" without changing the settings. Please close if this did the trick. :)

**Comment by groveld** [comment]
"System: Settings: General" and simply trigger "Save" without changing the settings didn't work for me, i had to change the firmware mirror temporarily for it to work.
Solved now!

**Comment by fichtner** [comment]
Found the root cause: https://github.com/opnsense/core/commit/8061ea1e5b9af23399e510dd87668ba06340cbff

thanks again :)

**Comment by module0x90** [comment]
It seems this is an issue with 15.7.20-amd64. I tried the following to fix it:

* "System: Settings: General" and simply trigger "Save"

* Change the mirror - I tried all and none of them fixed it.

* My /usr/local/etc/pkg/repos/origin.conf  looks like this:
OPNsense: {
  fingerprints: "/usr/local/etc/pkg/fingerprints/OPNsense",
  url: "pkg+http://pkg.opnsense.org/${ABI}/15.7/latest",
  signature_type: "fingerprints",
  mirror_type: "srv",
  priority: 11,
  enabled: yes
}
  Is this ok?

* Then I tried "manually" via the shell 
$ opnsense-update
Updating OPNsense repository catalogue...
OPNsense repository is up-to-date.
All repositories are up-to-date.
Updating OPNsense repository catalogue...
OPNsense repository is up-to-date.
All repositories are up-to-date.
Checking for upgrades (73 candidates): 100%
Processing candidates (73 candidates): 100%
Checking integrity... done (0 conflicting)
Your packages are up to date.
Checking integrity... done (0 conflicting)
Nothing to do.
Nothing to do.
Your system is up to date.

* It seems "12) Upgrade from console" does the same as opnsense-update.

What am I missing here?


Cheers

Thomas

PS: Keep up the good work, I love it!

**Comment by fichtner** [comment]
Hmm, can you please provide the output of:

    # pkg update -f
    # pkg upgrade -n
    # pkg info
    # ping pkg.opnsense.org

PS: First fix this, then carry on with the rest. :)

**Comment by module0x90** [comment]
```bash
$ pkg update -f
Updating OPNsense repository catalogue...
Fetching meta.txz: 100%    1 KiB   1.5kB/s    00:01    
Fetching packagesite.txz: 100%   73 KiB  74.5kB/s    00:01    
Processing entries: 100%
OPNsense repository update completed. 252 packages processed.

$ pkg upgrade -n
Updating OPNsense repository catalogue...
OPNsense repository is up-to-date.
All repositories are up-to-date.
Checking for upgrades (78 candidates): 100%
Processing candidates (78 candidates): 100%
The following 44 package(s) will be affected (of 0 checked):

Installed packages to be UPGRADED:
	suricata: 2.0.9 -> 2.0.10_1
	strongswan: 5.3.4 -> 5.3.5_1
	squid: 3.5.11 -> 3.5.12
	py27-setuptools27: 17.0 -> 18.7
	png: 1.6.19 -> 1.6.20
	php56-zlib: 5.6.15 -> 5.6.16
	php56-xml: 5.6.15 -> 5.6.16
	php56-tokenizer: 5.6.15 -> 5.6.16
	php56-sqlite3: 5.6.15 -> 5.6.16
	php56-sockets: 5.6.15 -> 5.6.16
	php56-simplexml: 5.6.15 -> 5.6.16
	php56-session: 5.6.15 -> 5.6.16
	php56-pdo_sqlite: 5.6.15 -> 5.6.16
	php56-pdo: 5.6.15 -> 5.6.16
	php56-openssl: 5.6.15 -> 5.6.16
	php56-mysql: 5.6.15 -> 5.6.16
	php56-mcrypt: 5.6.15 -> 5.6.16
	php56-mbstring: 5.6.15 -> 5.6.16
	php56-ldap: 5.6.15 -> 5.6.16
	php56-json: 5.6.15 -> 5.6.16
	php56-hash: 5.6.15 -> 5.6.16
	php56-gettext: 5.6.15 -> 5.6.16
	php56-filter: 5.6.15 -> 5.6.16
	php56-dom: 5.6.15 -> 5.6.16
	php56-curl: 5.6.15 -> 5.6.16
	php56-ctype: 5.6.15 -> 5.6.16
	php56-bz2: 5.6.15 -> 5.6.16
	php56-bcmath: 5.6.15 -> 5.6.16
	php56: 5.6.15 -> 5.6.16
	phalcon: 2.0.8 -> 2.0.9
	opnsense: 15.7.20 -> 15.7.22
	openssl: 1.0.2_4 -> 1.0.2_5
	openldap-client: 2.4.42_2 -> 2.4.43
	lighttpd: 1.4.37 -> 1.4.38
	curl: 7.45.0 -> 7.46.0_1

Installed packages to be REINSTALLED:
	syslogd-10.1_1 (option removed: ZMQ)
	pcre-8.37_4
	oniguruma4-4.7.1_1
	libyaml-0.1.6_2
	libnet-1.1.6_3,1
	libmcrypt-2.5.8_3
	libhtp-0.5.18
	jansson-2.7_1
	gettext-runtime-0.19.6

The process will require 199 KiB more space.
25 MiB to be downloaded.

$ pkg info
apinger-0.6.1_4                IP device monitoring tool
ataidle-2.7.2                  Utility to spin down ATA drives
beep-1.0_1                     Beeps a certain duration and pitch out of the PC Speaker
bind910-9.10.3_1               BIND DNS suite with updated DNSSEC and DNS64
bsdinstaller-2.3_5             BSD Installer mega-package
bsnmp-regex-0.6_1              bsnmpd module allowing creation of counters from log files
bsnmp-ucd-0.4.1                bsnmpd module that implements parts of UCD-SNMP-MIB
ca_root_nss-3.20.1             Root certificate bundle from the Mozilla Project
choparp-20150613               Simple proxy arp daemon
clog-1.0.1_3                   Circular log support for FreeBSD syslogd
cpdup-1.17_2                   Comprehensive filesystem mirroring and backup program
cpustats-0.1                   Gather system statistics
curl-7.45.0                    Non-interactive tool to get files from FTP, GOPHER, HTTP(S) servers
dhcp6-20080615_3               KAME DHCP6 client, server, and relay
dhcpleases-0.2                 Read dhpcd.lease file and add it to hosts file
dnsmasq-2.75,1                 Lightweight DNS forwarder, DHCP, and TFTP server
easy-rsa-2.2.2                 Small RSA key management package based on openssl
expat-2.1.0_3                  XML 1.0 parser written in C
expiretable-0.6_1              Utility to remove entries from the pf(4) table based on their age
filterdns-0.2                  filterdns
filterlog-0.2                  Parse pflog(4) output
freetype2-2.6_1                Free and portable TrueType font rendering engine
gettext-runtime-0.19.6         GNU gettext runtime libraries and programs
gmp-5.1.3_2                    Free library for arbitrary precision arithmetic
idnkit-1.0_5                   Library to handle internationalized domain names
ifinfo-10.1                    Interface statistics reader
igmpproxy-0.1_2,1              Multicast forwarding IGMP proxy
indexinfo-0.2.4                Utility to regenerate the GNU info page index
isc-dhcp42-client-4.2.8        The ISC Dynamic Host Configuration Protocol client
isc-dhcp42-relay-4.2.8         The ISC Dynamic Host Configuration Protocol relay
isc-dhcp42-server-4.2.8_1      ISC Dynamic Host Configuration Protocol server
jansson-2.7_1                  C library for encoding, decoding, and manipulating JSON data
ldns-1.6.17_5                  Library for programs conforming to DNS RFCs and drafts
libart_lgpl-2.3.21_2,1         Library for high-performance 2D graphics
libdaemon-0.14_1               Lightweight C library that eases the writing of UNIX daemons
libedit-3.1.20150325_1         Command line editor library
libevent2-2.0.22_1             API for executing callback functions on events or timeouts
libffi-3.2.1                   Foreign Function Interface
libhtp-0.5.18                  Security-aware parser for the HTTP protocol
libiconv-1.14_9                Character set conversion library
libltdl-2.4.6                  System independent dlopen wrapper
libmcrypt-2.5.8_3              Multi-cipher cryptographic library (used in PHP)
libnet-1.1.6_3,1               C library for creating IP packets
libpdel-0.5.3_6                Packet Design multi-purpose C library for embedded applications
libxml2-2.9.3                  XML parser library for GNOME
libyaml-0.1.6_2                YAML 1.1 parser and emitter written in C
lighttpd-1.4.37                Secure, fast, compliant, and flexible Web Server
lzo2-2.09                      Portable speedy, lossless data compression library
minicron-0.0.2                 A very small cron
miniupnpd-1.9_1,1              UPnP IGD implementation which uses pf/ipf
mpd4-4.4.1_2                   Multi-link PPP daemon based on netgraph(4)
mpd5-5.7_3                     Multi-link PPP daemon based on netgraph(4)
nettle-2.7.1                   Low-level cryptographic library
ntp-4.2.8p4                    The Network Time Protocol Distribution
oniguruma4-4.7.1_1             BSDL Regular Expressions library compatible with POSIX/GNU/Perl
openldap-client-2.4.42_2       Open source LDAP client implementation
openssh-portable-7.1.p1_4,1    The portable version of OpenBSD's OpenSSH
openssl-1.0.2_4                SSL and crypto library
openvpn-2.3.8                  Secure IP/Ethernet tunnel daemon
opnsense-15.7.20               OPNsense release package
opnsense-update-15.7.20        OPNsense update utility
pcre-8.37_4                    Perl Compatible Regular Expressions library
pecl-radius-1.2.7              Radius client library for PHP
perl5-5.20.3_8                 Practical Extraction and Report Language
pftop-0.7_6                    Utility for real-time display of statistics for pf
phalcon-2.0.8                  Phalcon PHP Framework written in C-language
php-pfSense-0.2_4              Library for getting useful info
php-suhosin-0.9.38             PHP extension that implements high-level protections
php-xdebug-2.2.5               Xdebug extension for PHP
php56-5.6.15                   PHP Scripting Language
php56-bcmath-5.6.15            The bcmath shared extension for php
php56-bz2-5.6.15               The bz2 shared extension for php
php56-ctype-5.6.15             The ctype shared extension for php
php56-curl-5.6.15              The curl shared extension for php
php56-dom-5.6.15               The dom shared extension for php
php56-filter-5.6.15            The filter shared extension for php
php56-gettext-5.6.15           The gettext shared extension for php
php56-hash-5.6.15              The hash shared extension for php
php56-json-5.6.15              The json shared extension for php
php56-ldap-5.6.15              The ldap shared extension for php
php56-mbstring-5.6.15          The mbstring shared extension for php
php56-mcrypt-5.6.15            The mcrypt shared extension for php
php56-mysql-5.6.15             The mysql shared extension for php
php56-openssl-5.6.15           The openssl shared extension for php
php56-pdo-5.6.15               The pdo shared extension for php
php56-pdo_sqlite-5.6.15        The pdo_sqlite shared extension for php
php56-session-5.6.15           The session shared extension for php
php56-simplexml-5.6.15         The simplexml shared extension for php
php56-sockets-5.6.15           The sockets shared extension for php
php56-sqlite3-5.6.15           The sqlite3 shared extension for php
php56-tokenizer-5.6.15         The tokenizer shared extension for php
php56-xml-5.6.15               The xml shared extension for php
php56-zlib-5.6.15              The zlib shared extension for php
pkg-1.6.2                      Package manager
png-1.6.19                     Library for manipulating PNG images
py27-Babel-2.1.1               Collection of tools for internationalizing Python applications
py27-Jinja2-2.8                Fast and easy to use stand-alone template engine
py27-MarkupSafe-0.23           Implements XML/HTML/XHTML Markup safe string for Python
py27-pytz-2015.7,1             World Timezone Definitions for Python
py27-requests-2.8.1            HTTP library written in Python for human beings
py27-setuptools27-17.0         Python packages installer
py27-sqlite3-2.7.10_7          Standard Python binding to the SQLite3 library (Python 2.7)
py27-ujson-1.33                Ultra fast JSON encoder and decoder for Python
python27-2.7.10_1              Interpreted object-oriented programming language
radvd-1.15                     Linux/BSD IPv6 router advertisement daemon
rate-0.9                       Traffic analysis command-line utility
relayd-5.5.20140810_1          OpenBSD relay daemon
rrdtool12-1.2.30_7             Round Robin Database Tools v1.2
smartmontools-6.4_1            S.M.A.R.T. disk monitoring tools
sqlite3-3.9.2                  SQL database engine in a C library
squid-3.5.11                   HTTP Caching Proxy
sshlockout_pf-0.0.2_2          Automatically block IPs with failed SSH logins using pf(4)
strongswan-5.3.4               Open Source IKEv2 IPsec-based VPN solution
sudo-1.8.15                    Allow others to run commands as root
suricata-2.0.9                 High Performance Network IDS, IPS and Security Monitoring engine
syslogd-10.1_1                 FreeBSD syslogd with additions
unbound-1.5.5                  Validating, recursive, and caching DNS resolver
voucher-0.1_4                  voucher support
wol-0.7.1_2                    Tool to wake up Wake-On-LAN compliant computers
zip-3.0_1                      Create/update ZIP files compatible with PKZIP

$ ping -c 2 pkg.opnsense.org
PING pkg.opnsense.org (37.48.77.141): 56 data bytes
64 bytes from 37.48.77.141: icmp_seq=0 ttl=54 time=34.692 ms
64 bytes from 37.48.77.141: icmp_seq=1 ttl=54 time=35.315 ms

--- pkg.opnsense.org ping statistics ---
2 packets transmitted, 2 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 34.692/35.004/35.315/0.311 ms
```

**Comment by module0x90** [comment]
That's interesting.

Now when I run that /usr/local/sbin/opnsense-update it seems to be working, ie. displaying more or less the same output as done with your individual commands.
The webgui though still insists on "Connection Error".

I am not too desparate to upgrade, but I really want to see the "new" webgui... :-) 

Would you want to run anymore commands why the webgui is still failing? I am all ears.
Should I reboot maybe, then try the webgui upgrade again?


Thomas

**Comment by fichtner** [comment]
Thanks, that looks perfectly fine. Try the GUI again now. :)

**Comment by fichtner** [comment]
Backend service may have failed, but I don't know. opnsense-update and the console are back to normal now for sure. I'd do console upgrade and reboot. The next GUI fetch should tell you it's all up to date then. If not there's something unique to your setup we're missing.

**Comment by module0x90** [comment]
I'll reboot now.
(Declaring downtime...)

Cheers

T

**Comment by module0x90** [comment]
Rebooted.
Same again in the webgui - but working from CLI.

Just a thought. The "Connection error" looks like a timeout, some sort. Which process is started when I click on "Click to check for updates"? Maybe that process has some sort-of problem - ?

**Comment by module0x90** [comment]
I managed to do packet captures on my WAN interface for "host pkg.opnsense.org" (and I needed a reason to try out Wireshark 2.0 with the Qt GUI... is really good.)

Scenario one: webgui
Downloads only http://pkg.opnsense.org/FreeBSD:10:amd64/15.7/latest/meta.txz
When adding "port domain" packets as well doesn't show any problems whatsoever, ie. neither DNS timeouts nor a non-responsive DNS server.

Scenario two: pkg update -f
Downloads http://pkg.opnsense.org/FreeBSD:10:amd64/15.7/latest/packagesite.txz as well, not just meta.txz.

**Comment by module0x90** [comment]
Still looking... not giving up yet:

$ /usr/local/opnsense/scripts/pkg_updatecheck.sh
{"connection":"error","repository":"error","last_check":"Wed Dec  9 23:05:50 GMT 2015","updates":"","download_size":"none","extra_space_required":"none","new_packages":[],"reinstall_packages":[],"upgrade_packages":[],"upgrade_needs_reboot":"0"}

Hm.

**Comment by module0x90** [comment]
OK, I think I found the problem - this can be closed now. Working for me. My setup.

I am running IPv6 with an Hurricane Electric IPv6 tunnel. Works great - the tunnel, not when you run OpenDNS as well. Somehow OpenDNS IPv6 addresses - Sandbox IPv6 DNS servers 2620:0:ccc::2 and 2620:0:ccd::2 - got into my config and that's it.

I managed to get as fas as /usr/local/opnsense/scripts/pkg_updatecheck.sh which takes 36s instead of the maximum 30s (as per webgui + script). When I changed that script timeout to 40s it was working alright, even with the faulty IPv6 DNS server.

Now I removed the fault IPv6 DNS server and WebGUI updates are working alright again.

Feel free to send IPv6 bugs to my direction... :-) I won't mind.

**Comment by fichtner** [comment]
Thanks for reporting back. 30 seconds seems overly long, maybe one of the two OpenDNS servers simply doesn't answer?

**Comment by module0x90** [comment]
Franco,

I explained all my findings on that bug report. Can be closed now, WONTFIX.

It seems my OPNSense installation has some wierd problem contacting some IPv6 addresses,
ping works, but not "dig @$IPV6" for example.

The IPV6 is a local issue - I will see to that, whether it is my config or a bug somewhere.
It doesn't cause any issues really.

Updated to .22 yesterday night and the new interface is great! Thanks.
BR

Thomas
December 10 2015 2:35 PM, "Franco Fichtner"  wrote:
	Thanks for reporting back. 30 seconds seems overly long, maybe one of the two OpenDNS servers simply doesn't answer?
